src/overlays/actors/ovl_En_Rr/z_en_rr.c

```diff
@@ -21,51 +81,772 @@ const ActorInit En_Rr_InitVars = {
+        this->collider1.base.acFlags &= ~2;
+    }
+    func_8002E4B4(globalCtx, &this->actor, 20.0f, 30.0f, 20.0f, 7);
+    if (!this->stopScroll) {
+        Math_SmoothScaleMaxF(&this->pulsePhaseVel, this->phaseVelTarget, 1.0f, 50.0f);
+        Math_SmoothScaleMaxF(&this->pulseRate, 4.0f, 1.0f, 5.0f);
+        Math_SmoothScaleMaxF(&this->wobbleRateX, this->wobbleRateXTarget, 1.0f, 0.04f);
+        Math_SmoothScaleMaxF(&this->wobbleRateZ, this->wobbleRateZTarget, 1.0f, 0.01f);
+        Math_SmoothScaleMaxF(&this->pulseSize, this->pulseSizeTarget, 1.0f, 0.0015f);
+        Math_SmoothScaleMaxF(&this->wobbleSize, this->wobbleSizeTarget, 1.0f, 20.0f);
+        for (i = 0; i < 5; i++) {
+            Math_SmoothScaleMaxMinS(&this->bodySegs[i].rot.x, this->bodySegs[i].rotTarget.x, 5, this->unk_364 * 1000.0f,
+                                    0);
+            Math_SmoothScaleMaxMinS(&this->bodySegs[i].rot.z, this->bodySegs[i].rotTarget.z, 5, this->unk_364 * 1000.0f,
+                                    0);
+            Math_SmoothScaleMaxF(&this->bodySegs[i].scale.x, this->bodySegs[i].scaleTarget.x, 1.0f,
+                                 this->unk_364 * 0.2f);
+            this->bodySegs[i].scale.z = this->bodySegs[i].scale.x;
+            Math_SmoothScaleMaxF(&this->bodySegs[i].height, this->bodySegs[i].heightTarget, 1.0f,
+                                 this->unk_364 * 300.0f);
+        }
+        Math_SmoothScaleMaxF(&this->unk_364, 1.0f, 1.0f, 0.2f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE978C.s")
+void EnRr_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f zeroVec;
+    EnRr* this = THIS;
+    s32 i;
+    Mtx* segMats = Graph_Alloc(globalCtx->state.gfxCtx, 4 * sizeof(Mtx));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE9880.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_rr.c", 1478);
+    if (1) {}
+    func_80093D84(globalCtx->state.gfxCtx);
+    gSPSegment(POLY_XLU_DISP++, 0x0C, segMats);
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (this->scrollTimer * 0) & 0x7F,
+                                (this->scrollTimer * 0) & 0x3F, 32, 16, 1, (this->scrollTimer * 0) & 0x3F,
+                                (this->scrollTimer * -6) & 0x7F, 32, 16));
+    Matrix_Push();
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE98F8.s")
+    Matrix_Scale((1.0f + this->bodySegs[RR_BASE].scaleMod.x) * this->bodySegs[RR_BASE].scale.x,
+                 (1.0f + this->bodySegs[RR_BASE].scaleMod.y) * this->bodySegs[RR_BASE].scale.y,
+                 (1.0f + this->bodySegs[RR_BASE].scaleMod.z) * this->bodySegs[RR_BASE].scale.z, 1);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_rr.c", 1501),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    Matrix_Pull();
+    VEC_SET(zeroVec, 0.0f, 0.0f, 0.0f);
```

Comment:
had to be sameline to match?

Reply:
Probably not. Should `VEC_SET` only be used if needed?

Reply:
for now yeah, we have very few uses of it and it was only created for sameline purposes.
imo the macro probably didnt even exist and it was just written on the same line sometimes. but we can decide if we want to use it more later